### DISCUSSION TITLE
why the txs in block are not ordered by gas price

### DISCUSSION THREAD

**Discussion by crazybits** [aptos-labs/aptos-developer-discussions#404] [Questions] [answered]
### Describe your question in detail.

refer to the block [222784980](https://aptoscan.com/block/222784980),the txs ordering in block are not ordered by gas price, why this happened?

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

Rust SDK

**Comment by ying-w** [comment, marked as answer]
> higher gas fees only prioritize transaction selection for the next block.
> However, within a block, the order of transaction execution is determined by the system.

https://aptos.dev/en/network/blockchain/gas-txn-fee#gas-price-and-prioritizing-transactions
